src/server/services/meta_service.h

```diff
@@ -618,6 +620,42 @@ class IMetaService {
     }
   }
 
+  /// Flatten potential subtree, which is uploaded to etcd by `Persist`.
+  /// See meta_tree.cc:`generate_persist_ops` for reference.
+  void decode_ops(const IMetaService::op_t& op,
+                  std::vector<IMetaService::op_t>& out) {
+    auto kv = op.kv;
+    if (kv.key.back() == '.' && op.op == op_t::op_type_t::kPut) {
+      boost::property_tree::ptree pt;
+      std::stringstream ss(kv.value);
+      boost::property_tree::read_json(ss, pt);
+
+      for (ptree::iterator it = pt.begin(); it != pt.end(); ++it) {
+        std::string encoded_value;
+        if (it->second.empty()) {
+          if (it->first == "id" || it->second.data().empty()) {
+            continue;
+          }
+          encoded_value = "v";
+          if (it->first == "transient") {
+            encoded_value += "false";
+          } else {
+            encoded_value += it->second.data();
```

Comment:
Use `encode_value()`.

Reply:
Having a `__subtree` can help to recover the link, otherwise the link would take up an additional record. This way the record number can be halved.